OCPBUGS-17088: Baremetal IPI install doc is missing required port access information (#63913)
openshift/openshift-docs: modules/ipi-install-network-requirements.adoc

image::210_OpenShift_Baremetal_IPI_Deployment_updates_0122_2.png[Installer-provisioned networking]

[id="network-requirements-ensuring-required-ports-are-open_{context}"]
== Ensuring required ports are open

Certain ports must be open for installer-provisioned installations to compelete successfully. In most cases, the installer will configure this for you. In certain situations, such as using separate subnets for far edge worker nodes, you must ensure that worker nodes can communicate with control plane nodes over these ports.

.Required ports
[options="header"]
|====
|Port|Description

> |`5050`, `5051`| The Ironic Inspector API runs on the control plane nodes and listens on port `5050` and on secured port `5051`. The Inspector API is responsible for hardware introspection, which collects information about the hardware characteristics of the bare metal hosts.

Reviewer: 5051 is a private port, let's not mention it

> |`6180`, `6183`| The baseboard management controllers (BMC) of the cluster nodes must be able to access the control plane nodes using port `6180` or secured port `6183`.

Reviewer: let's say that 6183 is the preferred option and this is needed to get the inspection and deployment images

> |`6385`, `6388`| The Ironic API server runs intially on the bootstrap VM and later on the control plane nodes and listens on port `6385` and secured port `6388`. The Ironic API allows clients to interact with Ironic for bare metal host provisioning and management, including operations like enrolling new hosts, managing their power state, deploying images, and cleaning the hardware.

Reviewer: 6388 is a private, let's not mention it
6385 is used as either secured or not, the secure option is the preferred one

image::210_OpenShift_Baremetal_IPI_Deployment_updates_0122_2.png[Installer-provisioned networking]

[id="network-requirements-ensuring-required-ports-are-open_{context}"]
== Ensuring required ports are open

Certain ports must be open for installer-provisioned installations to compelete successfully. In most cases, the installer will configure this for you. In certain situations, such as using separate subnets for far edge worker nodes, you must ensure that worker nodes can communicate with control plane nodes over these ports.

.Required ports
[options="header"]
|====
|Port|Description

|`5050-5051`| The Ironic Inspector API runs on the control plane nodes and listens on ports `5050` and `5051`. The Inspector API is responsible for hardware introspection, which collects information about the hardware characteristics of the bare metal hosts.

> |`6180`| The baseboard management controllers (BMC) of the cluster nodes must be able to access the control plane nodes using port `6180`.

Reviewer: 6180 and 6183 give access to the same backend, it's just the former is not secured, the latter is

|`6183`| When installing with virtual media, Baseboard Management Controllers (BMC) retrieve the virtual media image from the server hosting the virtual media image by requesting it using port `6183` by default.

> |`6385`| The Ironic API server runs intially on the bootstrap VM and later on the control plane nodes and listens on port `6385` by default. The Ironic API allows clients to interact with Ironic for bare metal host provisioning and management, including operations like enrolling new hosts, managing their power state, deploying images, and cleaning the hardware.

Reviewer: I would mention 6388 as well with 6385

image::210_OpenShift_Baremetal_IPI_Deployment_updates_0122_2.png[Installer-provisioned networking]

[id="network-requirements-ensuring-required-ports-are-open_{context}"]
== Ensuring required ports are open

> Certain ports must be open between cluster nodes for installer-provisioned installations to compelete successfully. In certain situations, such as using separate subnets for far edge worker nodes, you must ensure that the nodes in these subnets can communicate with nodes in other subnets on the following required ports.

Reviewer 2: ```suggestion
Certain ports must be open between cluster nodes for installer-provisioned installations to complete successfully. In certain situations, such as using separate subnets for far edge worker nodes, you must ensure that the nodes in these subnets can communicate with nodes in other subnets on the following required ports.
```

.Required ports
[options="header"]
|====
|Port|Description

|`67`,`68` | When using a provisioning network, cluster nodes access the `dnsmasq` DHCP server over their provisioning network interfaces using ports `67` and `68`.

> | `69` | When using a provisioning network, cluster nodes communicate with the TFTP server on port `69` using their provisioning network interfaces. The TFTP server runs on the bootrap VM. The bootstrap VM runs on the provisioner node.

Reviewer 2: ```suggestion
| `69` | When using a provisioning network, cluster nodes communicate with the TFTP server on port `69` using their provisioning network interfaces. The TFTP server runs on the bootstrap VM. The bootstrap VM runs on the provisioner node.
```

| `123` | The cluster nodes must access the NTP server on port `123` using the `baremetal` machine network.

|`5050`| The Ironic Inspector API runs on the control plane nodes and listens on port `5050`. The Inspector API is responsible for hardware introspection, which collects information about the hardware characteristics of the bare metal hosts.

|`6180`| When deploying with virtual media and not using TLS, the provisioner node and the control plane nodes must have port `6180` open on the `baremetal` machine network interface so that the baseboard management controller (BMC) of the worker nodes can access the {op-system} image. Starting with {product-title} 4.13, the default HTTP port is `6180`.

|`6183`| When deploying with virtual media and using TLS, the provisioner node and the control plane nodes must have port `6183` open on the `baremetal` machine network interface so that the BMC of the worker nodes can access the {op-system} image.

> |`6385`| The Ironic API server runs intially on the bootstrap VM and later on the control plane nodes and listens on port `6385`. The Ironic API allows clients to interact with Ironic for bare metal host provisioning and management, including operations like enrolling new hosts, managing their power state, deploying images, and cleaning the hardware.

Reviewer 2: ```suggestion
|`6385`| The Ironic API server runs initially on the bootstrap VM and later on the control plane nodes and listens on port `6385`. The Ironic API allows clients to interact with Ironic for bare metal host provisioning and management, including operations like enrolling new hosts and managing their power state, deploying images, and cleaning the hardware.
```

[id='network-requirements-out-of-band_{context}']
== Port access for the out-of-band management IP address

> The out-of-band management IP address is on a separate network from the node. To ensure that the out-of-band management can communicate with the provisioner node during installation, the out-of-band management IP address must be granted access to port `6180` on the provisioner node and on the {product-title} control plane nodes. TLS port `6183` is required for virtual media installation, for example, via Redfish.

Reviewer 2: ```suggestion
The out-of-band management IP address is on a separate network from the node. To ensure that the out-of-band management IP address can communicate with the provisioner node during installation, the out-of-band management IP address must be granted access to port `6180` on the provisioner node and on the {product-title} control plane nodes. TLS port `6183` is required for virtual media installation, for example, by using Redfish.
```

Reviewer 2: Per IBM Style, "via" should only be used to refer to routing data. https://www.ibm.com/docs/en/ibm-style?topic=word-usage#v
